include/LightningJSON/Exceptions.hpp

```diff
@@ -48,6 +60,7 @@ class LightningJSON::JSONTypeMismatch : public JSONException
 			case JSONType::Object:  return "JSON type mismatch: Expected 'integer', but the type found was 'object'.";
 			case JSONType::Double:  return "JSON type mismatch: Expected 'integer', but the type found was 'double'.";
 			case JSONType::Null:    return "JSON type mismatch: Expected 'integer', but the type found was 'null'.";
+			case JSONType::Integer: abort(); // This case should never happen.
```

Comment:
This seems very wrong. If m_expectedType equals m_foundType, abort?! I think this should just return "Malformed exception: types match"

Reply:
That seemed to me a case that should never happen, meaning a bug in the library. Aborting is just what I do in that case. I can return a string instead.

Reply:
I agree it shouldn't happen, but if it DOES happen, aborting here (from the end user perspective) is taking a minor bug in the library and turning it into an unrecoverable program-halting bug in the library. I think flagging the error without aborting the whole process is better in this case.